### PR TITLE
Santizes custom render function.

### PR DESCRIPTION
**Pull request by goodroot**

### CODE REVIEW

File: custom.html [hunk 1]
```diff
@@ -8,16 +8,17 @@
...
-            out = out.concat('<p class="url">' + item['author'] + '</p>');
+            var out = '<p class="title">' + item.highlight['title'] + '</p>';
+            out = out.concat('<p class="url">' + item.highlight['author'] + '</p>');
             return out;
           };
 
           $('#st-search-input').swiftype({
             engineKey: 'iZFpjpfr3fdLXgpn71rv',
             renderFunction: customRenderFunction,
+            highlightFields: {{'books': {'title': {'size': 60, 'fallback': true}, 'author': {'size': 30, 'fallback': true}}},
```
Comment (JasonStoltz) [inline custom.html]: Looks like there is an error here. Try this:

```
diff --git a/custom.html b/custom.html
index 8c82962..dc68e65 100644
--- a/custom.html
+++ b/custom.html
@@ -16,7 +16,7 @@
           $('#st-search-input').swiftype({
             engineKey: 'iZFpjpfr3fdLXgpn71rv',
             renderFunction: customRenderFunction,
-            highlightFields: {{'books': {'title': {'size': 60, 'fallback': true}, 'author': {'size': 30, 'fallback': true}}},
+            highlightFields: {'books': {'title': {'size': 60, 'fallback': true}, 'author': {'size': 30, 'fallback': true}}},
             searchFields: {'books': ['title'], 'magazines': ['name']},
             fetchFields: {'books': ['price']},
             filters: function() {
```

File: README.md [hunk 2]
```diff
@@ -78,21 +77,22 @@ Now that you have more data for each autocomplete item, you'll want to customize
 The default rendering function is shown below:
 
 	var defaultRenderFunction = function(document_type, item) {
```
Comment (JasonStoltz) [inline README.md]: Interestingly, the `defaultRenderFunction` listed is incorrect. It's actually:

```
var defaultRenderFunction = function(document_type, item) {
    return '<p class="title">' + Swiftype.htmlEscape(item['title']) + '</p>';
  };
```

So, that's another way to sanitize your data. Anywhere you're doing `item.highlight[]`, you COULD do `Swiftype.htmlEscape(item[])`. Whichever the preferred solution is.

File: README.md [hunk 3]
```diff
@@ -78,21 +77,22 @@ Now that you have more data for each autocomplete item, you'll want to customize
...
-		return '<p class="title">' + item['title'] + '</p>';
+		return '<p class="title">' + item.highlight['title'] + '</p>';
 	};
 
 The additional fields are available as keys in the item dictionary, so you could customize this to make use of the `genre` field as follows:
 
 	var customRenderFunction = function(document_type, item) {
-		var out = '<p class="title">' + item['title'] + '</p>';
-		return out.concat('<p class="genre">' + item['genre'] + '</p>');
+	  var out = '<a href="' + item['url'] + '" class="st-search-result-link">' + item.highlight['title'] + '</a>';
```
Comment (JasonStoltz) [inline README.md]: Ah, actually, here's a perfect example. Using `item['url']` here IS still actually an XSS vulnerability.

So you could change to:
`Swiftype.htmlEscape(item['url'])`

File: custom_dropdown.html [hunk 4]
```diff
@@ -38,11 +38,12 @@
           };
 
           $('#st-search-input').swiftype({
-            engineKey: 'V7U1B7opm5xDnVFnsk5d',
+            engineKey: '5jZG1gmmCTFYbSSDjpqq',
```
Comment (JasonStoltz) [inline custom_dropdown.html]: @goodroot Is this engine key supposed to be in here? Just saw that it changed, wanted to make sure it was intentional.

Comment (goodroot) [reply custom_dropdown.html]: @JasonStoltz Yes -- the prior key is a... strange key.

**Pull request merged**